addons/website_sale_comparison/controllers/main.py

```diff
@@ -19,13 +20,15 @@ def product_compare(self, **post):
 
         res = {}
         for num, product in enumerate(products):
-            for var in product.attribute_line_ids:
-                cat_name = var.attribute_id.category_id.name or _('Uncategorized')
-                att_name = var.attribute_id.name
+            attr_ids = product.attribute_line_ids.mapped('attribute_id.id')
+            # to get ordered recordset we have to use search
+            for attr in request.env['product.attribute'].search([('id', 'in', attr_ids)]):
```

Comment:
why not use :
`for attr in product.attribute_line_ids.mapped('attribute_id').sorted(key=lambda x: (x.sequence, x.name)):`
?

Reply:
Well, at first I did it exactly like that but then I thought that I hardcoded ordering logic. If it will be changed on model definition it should be changed here too. So I went for `search` to inherit whatever `_order` is defined on the model.

Reply:
in this case you can write this

`for attr in product.attribute_line_ids.mapped('attribute_id').sorted():`

because : https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4555

Reply:
That's sweet